### PR TITLE
add test a solidity event tutorial files

### PR DESCRIPTION
**Pull request by a-ridley**
### Description
 This short tutorial teaches developers how to test a solidity event using foundry cheatcodes.
 The tutorial text is located here : https://github.com/hedera-dev/preview-hedera-docs/pull/12/commits/3cfa42636ede1d30b32612452edc8d2d1aebdacb

### What is being added
Foundry manages dependencies through gitsubmodules so a `.gitsubmodule` file is added along with the directory, `lib`, that contains the submodules.
Readme.md file which introduces the tutorial and provides a link to the tutorial on docs.hedera.com
A `src` folder which is where foundry looks for smart contracts
The `TodoList.sol` contract which we will be testing.
A `test` folder to house our solidity tests
The `TodoList.t.sol` test contract.
A `package.json` file which has the hashgraph SDK JS

### CODE REVIEW

File: test/TodoList.t.sol
Comment (ed-marquez) [inline test/TodoList.t.sol:1]: Fix typo in line 9: "expected"

Comment (ed-marquez) [reply test/TodoList.t.sol:1]: Also capitalize the comment instructions for steps 4 and 5

File: foundry.toml
Comment (ed-marquez) [inline foundry.toml:1]: Similar to part 1, be sure to specify the location where `gas_reports = ["TodoList"]` should be added in the file. 
If added below `[rpc_endpoints]`, the user will see errors.

**Review by ed-marquez — APPROVED** [review]
Code looks good and the tutorial is easy to follow and completes successfully.

image

Left a few comments in the PR about additional details and minor fixes in the code comments, but I'd consider this reviewed and ready for next steps.

File: foundry.toml
Comment (a-ridley) [reply foundry.toml:1]: oo good catch, this step 5 is from the previous tutorial. In this one, we aren't completing gas reports so there isn't a step. I will remove this from the toml file.

Comment (a-ridley) [reply foundry.toml:1]: i'll make sure the first tutorial has that step clear.

**Pull request merged**